mlflow/langchain/langchain_tracer.py

```diff
@@ -376,19 +377,28 @@ def on_tool_start(
         """Start span for a tool run."""
         if metadata:
             kwargs.update({"metadata": metadata})
+
+        # Input string should be JSON-like (it is json with single quotes)
+        # for function calling. We try parsing it for better rendering,
+        # but conservatively fallback to original if it fails.
+        try:
+            inputs = json.loads(input_str.replace("'", '"'))
```

Comment:
any chance that input_str includes single quotes that should be preserved and this line doesn't fail?

Reply:
`ast.literal_eval` makes sense!